docs/developer-guide/extensions/ui-extensions.md

```diff
@@ -158,3 +158,64 @@ Below is an example of an extension using the flyout widget:
   );
 })(window);
 ```
+
+## Top Bar Action Menu Extensions
+
+The top bar panel is the action menu at the top of the application view where the action buttons are displayed like Details, Sync, Refresh. Argo CD allows you to add new button to the top bar action menu of an application.
+When the extension button is clicked, the custom widget will be rendered in a flyout panel.
+
+The extension should be registered using the `extensionsAPI.registerTopBarActionMenuExt` method:
+
+```typescript
+registerTopBarActionMenuExt(
+        component: TopBarActionMenuExtComponent,
+        title: string,
+        id: string,
+        flyout?: ExtensionComponent,
+        shouldDisplay: (app?: Application) => boolean = () => true,
+        iconClassName?: string,
+        isMiddle = false
+)
+```
+Developer need to pass the function `shouldDisplay` which will be called to determine if the extension should be displayed or not. 
+The function should return true if the extension should be displayed and false otherwise.
```

Comment:
```suggestion

The callback function `shouldDisplay` should return true if the extension should be displayed and false otherwise:
```

- add a new line after the code block, per markdownlint
- simplify the text here which had several redundant phrases, per [k8s style guide](https://kubernetes.io/docs/contribute/style/style-guide/#use-simple-and-direct-language)
- add a colon to segue into the next code block